### ISSUE TITLE
train on my own datasets, the loss is nan

### ISSUE THREAD

**Issue by visonpon** [Kai-46/nerfplusplus#10]
Hi @Kai-46, after using colmap to get the pose and intrinsics of my own datasets and train from scratch,  the loss got nan, and I print the network's output, the ret['rgb']'s values are all nan.

I wonder whether the pose and intrinsics are wrong(`data[key]['K']` stores the intrinsic and `data[key]['W2C']` pose, right? )or I need to adjust some hyperparameters of the training phase?

Hope you can help, thanks~

**Comment by visonpon** [comment]
I have checked the colmap's output, when  I use the tanks_and_temples datasets you provided to test, the intrinsics of the output are the same as the intrinsics provided but pose not the same.

So, I wonder if there are some bugs in the` run_colmap.py`, hope can get some advice, thanks~

**Comment by Kai-46** [comment]
It's likely that not all your cameras are bounded by the unit sphere,  hence the [intersect_sphere function] (https://github.com/Kai-46/nerfplusplus/blob/acc089a449866fd71bc26975ca9ba67d1abfbb5a/ddp_train_nerf.py#L42) throws nan.

**Comment by visonpon** [comment]
after using the `raise exception`, it seems my camera poses are not normalized well indeed. Hope you can tell me how to normalize it, thanks~

**Comment by Kai-46** [comment]
It's well expected that SfM system can have different outputs, as there's an inherent ambiguity in pose estimation: you can choose your world coordinate system, in fact, size of the scene as well, arbitrarily.

**Comment by Kai-46** [comment]
FYI, I just added an interactive viewer for visually inspecting the normalized cameras. Hope it helps.

**Comment by visonpon** [comment]
Hi, @Kai-46, thanks for your generous share, but when I try to normalize my own camera parameters, I got a normed result like below

![24476195](https://user-images.githubusercontent.com/11517731/108977734-d2119500-76c3-11eb-98df-a34bedc67e73.jpg)

the original camera path(blue) is normal, but after normalize(green for target_radius=1.0, red for target_radius=0.7), the camera path are all changed.

Hope you can help, thanks~

**Comment by jly0810** [comment]
> after using the `raise exception`, it seems my camera poses are not normalized well indeed. Hope you can tell me how to normalize it.
> 
> And I have test `run_colmap.py` using the tanks_and_temples datasets and get intrinsic and pose for each image, while the intrinsics are the same as provided but the produced poses are not the same as provided in the datasets...

Have you solved this problem? I have encountered the same problem and it is very bothering me. Thank you very much
